.github/workflows/codacy-analysis.yml

```diff
@@ -0,0 +1,30 @@
+name: Codacy Analysis
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - "**.go"
+  pull_request:
+    branches: [main]
+    paths:
+      - "**.go"
+
+jobs:
+  codacy-analysis:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Run Codacy Analysis
+        uses: codacy/codacy-analysis-cli-action@97bf5df3c09e75f5bcd72695998f96ebd701846e
+        with:
+          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+
+      - name: Run Codacy Coverage Script
+        run: |
+          go test -coverprofile=unit.coverage.out ./...
+          bash <(curl -Ls https://coverage.codacy.com/get.sh) report \
```

Comment:
I think you can use their coverage reporter action as well: https://github.com/codacy/codacy-coverage-reporter-action